dds/DCPS/transport/framework/NetworkAddress.h

```diff
@@ -63,9 +65,15 @@ struct OpenDDS_Dcps_Export NetworkAddress {
 /// address. In the case using short hostname or name resolved from loopback, a
 /// warning is logged. If there is no any name discovered from network interfaces,
 /// an error is logged.
+/// If ACE_HAS_IPV6, will give priority to IPV6 interfaces
 extern OpenDDS_Dcps_Export
 OPENDDS_STRING get_fully_qualified_hostname(ACE_INET_Addr* addr = 0);
 
+/// Helper function to set the ttl on a socket appropriately
+/// given whether it is IPV4 or IPV6
+extern OpenDDS_Dcps_Export
+bool set_socket_ttl(const ACE_SOCK_Dgram& unicast_socket, const char& ttl);
```

Comment:
I think the use of 'char' for TTL has been a mistake from the start (it could just be int), but we need to consult documentation for ACE, linux manpages and MSDN.